modules/internal/DefaultAssociative.chpl

```diff
@@ -279,8 +279,29 @@ module DefaultAssociative {
       return _findFilledSlot(idx)(1);
     }
   
-    proc dsiAdd(idx: idxType, in slotNum : index(tableDom) = -1, haveLock = !parSafe): index(tableDom) {
+    proc dsiAdd(idx){
+      // add helpers will return a tuple like (slotNum, numIndicesAdded);
+
+      // these two seemingly redundant lines were necessary to work around a
+      // compiler bug. I was unable to create a smaller case that has the same
+      // issue.
+      // More: `return _addWrapper(idx)[2]` Call to _addWrapper seems to
+      // have no effect when `idx` is a range and the line is promoted. My
+      // understanding of promotion makes me believe that things might go haywire
+      // since return type of the method becomes an array(?). However, it seemed
+      // that _addWrapper is never called when the return statement is promoted.
+      // I checked the C code and couldn't see any call to _addWrapper.
+      // I tried to replicate the issue with generic classes but it always
+      // worked smoothly.
+      const numInds = _addWrapper(idx)[2];
```

Comment:
Did the problem occur with --baseline ? That might help us narrow down the issue.